pkg/k8s/endpoints.go

```diff
@@ -414,7 +414,17 @@ func (es *EndpointSlices) GetEndpoints() *Endpoints {
 	allEps := newEndpoints()
 	for _, eps := range es.epSlices {
 		for backend, ep := range eps.Backends {
-			allEps.Backends[backend] = ep
+			// EndpointSlices may have duplicate addresses on different slices.
+			b, ok := allEps.Backends[backend]
+			if !ok {
+				allEps.Backends[backend] = ep
+			} else {
+				clone := b.DeepCopy()
+				for k, v := range ep.Ports {
+					clone.Ports[k] = v
```

Comment:
Why do you explicitly copy over ports? Looks like `DeepCopy` should do the job, no?

Reply:
you need to merge the ports from two different backends with the same key without mutating the object.
You clone first backend and append the pods of the second backend ... the bug is because it overwrites the existing backends

Reply:
While this handles the case where endpoints with overlapping addresses but different ports are duplicated in ep slices, this doesn't handle the case where an endpoint as a whole is duplicated in ep slices, does it? If I interpreted this case described here correctly - https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/#duplicate-endpoints, won't we end up duplicating ports in the cloned backend?

Reply:
```
// PortConfiguration is the L4 port configuration of a frontend or backend. The
// map is indexed by the name of the port and the value constains the L4 port
// and protocol.
//
// +deepequal-gen=true
type PortConfiguration map[string]*loadbalancer.L4Addr
```
it is a map, it can't have duplicates